components/bl-mention-component/src/index.js

```diff
@@ -0,0 +1,124 @@
+import { useEffect, useState } from 'react';
+
+import primereact from './lib/core';
+
+const { Mention } = primereact.mention;
+const { cn } = BackendlessUI.CSSUtils;
+
+const DEFAULT_FIELD_BLACK_LIST = ['created', '___class', 'ownerId', 'updated', 'objectId'];
+
+export default function MentionComponent({ component, eventHandlers }) {
+  const {
+    trigger, suggestions, field, scrollHeight, autoHighlight, placeholder, delay, autoresize, rows, cols,
+    hideField, classList, style, display,
+  } = component;
+  const { onChange, onFocus, onBlur, onShow, onHide } = eventHandlers;
+
+  const [processedSuggestions, setProcessedSuggestions] = useState([]);
+  const triggers = useTriggers(trigger);
+  const [fieldBlackList, setFieldBlackList] = useState(DEFAULT_FIELD_BLACK_LIST);
+
+  useEffect(() => {
+    if (hideField?.length) {
+      const hideValues = hideField.split(',').map(el => el.trim());
+      setFieldBlackList(prev => [...prev, ...hideValues]);
+    }
+  }, [hideField]);
+
+  const searchHandler = event => {
+    const eventTrigger = event.trigger;
+
+    for (const trigger of triggers) {
+      if (trigger !== eventTrigger) {
+        continue;
+      }
+
+      const query = event.query;
+      const orderedSuggestions = suggestions.filter(suggestion => suggestion.trigger === eventTrigger)[0].suggestions;
```

Comment:
orderedSuggestions => suggestionsByTrigger(?)

Not sure, but it could be a map
```
{
 '@' : <list of suggestions>
 '#' : <list of suggestions>
}
```
What do you think?